### PR TITLE
PSPageControlProtocol and example applying the new protocol

### PR DESCRIPTION
**Pull request by tecnobeto**
- CurrentViewIndex now is public and has didSet
- PSPageControlProtocol added to get when CurrentViewIndex changed
- Example applying the new PSPageControlProtocol

### CODE REVIEW

File: PSPageControl.swift [hunk 1]
```diff
@@ -100,9 +107,21 @@ open class PSPageControl: UIView {
     fileprivate var background = UIImageViewAligned()
     fileprivate var pageControl = UIPageControl()
     fileprivate var touchPosition: CGPoint?
-    fileprivate var currentViewIndex = 0
     fileprivate var backgroundLayerFrameOrigin: CGPoint?
     
+    /// Delegate to detect when current PSPageControl changed
+    open var delegate: PSPageControlProtocol?
```
Comment (sochalewski) [inline PSPageControl.swift]: I believe it should be a `weak` variable.

File: PSPageControl.swift [hunk 2]
```diff
@@ -10,6 +10,13 @@ import AVFoundation
 import UIKit
 import UIImageViewAlignedSwift
 
+/**
+ Event to detect when PSPageControl was changed
+ */
+public protocol PSPageControlProtocol {
```
Comment (sochalewski) [inline PSPageControl.swift]: `PSPageControlDelegate` fits better naming convention.

**Review by sochalewski — CHANGES_REQUESTED** [review]
Thank you for your PR! I added these two comments, so after these changes I'll easily merge it.

**Comment by tecnobeto** [comment]
Ok, thanks for answer I will send a new pull request for you!

**Pull request closed without merging**